kernel/syscall.c

```diff
@@ -3126,6 +3126,37 @@ long myst_syscall_mbind(
     return ret;
 }
 
+long myst_syscall_get_mempolicy(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    unsigned long size = 0;
+    int ROUND_UP_VAL = sizeof(unsigned long) * 8;
+    myst_round_up(maxnode, ROUND_UP_VAL, &size);
+
```

Comment:
Suggest adding some input validation (the case that returns `EINVAL` as follows):

>        EINVAL The value specified by maxnode is less than the number of
>              node IDs supported by the system.  Or flags specified
>              values other than MPOL_F_NODE or MPOL_F_ADDR; or flags
>              specified MPOL_F_ADDR and addr is NULL, or flags did not
>              specify MPOL_F_ADDR and addr is not NULL.  Or, flags
>              specified MPOL_F_NODE but not MPOL_F_ADDR and the current
>              thread policy is not MPOL_INTERLEAVE.  Or, flags specified
>              MPOL_F_MEMS_ALLOWED with either MPOL_F_ADDR or
>              MPOL_F_NODE.  (And there are other EINVAL cases.)

Reply:
Added more testcases, thank you for pointing this out